commands.go

```diff
@@ -0,0 +1,746 @@
+}
+
+// NewDeleteDatabaseCommand creates a new DeleteDatabaseCommand.
+// fromNode can be empty string
+// https://sourcegraph.com/github.com/ravendb/RavenDB-Python-Client@v4.0/-/blob/pyravendb/raven_operations/server_operations.py#L79
+// TODO: add timeToWaitForConfirmation time.Duration? Don't know the format,
+// python example sends null
+func NewDeleteDatabaseCommand(dbName string, hardDelete bool, fromNode string) *RavenCommand {
+
+	opts := map[string]interface{}{}
+	opts["DatabaseNames"] = []string{dbName}
+	opts["HardDelete"] = hardDelete
+	if fromNode != "" {
+		opts["FromNodes"] = []string{fromNode}
+	}
+
+	data, err := json.Marshal(opts)
+	must(err)
+	res := &RavenCommand{
+		Method:      http.MethodDelete,
+		URLTemplate: "{url}/admin/databases",
+		Data:        data,
+	}
+	return res
+}
+
+// ExecuteDeleteDatabaseCommand executes CreateDatabaseCommand
+func ExecuteDeleteDatabaseCommand(exec CommandExecutorFunc, cmd *RavenCommand) (*DeleteDatabaseResult, error) {
+	var res DeleteDatabaseResult
+	err := excuteCmdAndJSONDecode(exec, cmd, &res)
+	if err != nil {
+		return nil, err
+	}
+	return &res, nil
+}
+
+// GetOperationStateCommandResult describes a result of GetOperationsState
+type GetOperationStateCommandResult struct {
+	ErrorStr string `json:"Error"`
+	Status   string `json:"Status"`
+}
+
+// NewGetOperationStateCommand creates GetOperationsState command
+// https://sourcegraph.com/github.com/ravendb/RavenDB-Python-Client/-/blob/pyravendb/commands/raven_commands.py#L371
+// https://sourcegraph.com/github.com/ravendb/ravendb-jvm-client/-/blob/src/main/java/net/ravendb/client/documents/operations/GetOperationStateOperation.java#L14
+// TODO: add isServerStoreOp bool? Is in Python, not in Java
+func NewGetOperationStateCommand(opID string) *RavenCommand {
+	res := &RavenCommand{
+		Method:      http.MethodGet,
+		URLTemplate: "{url}/operations/state?id=" + opID,
+	}
+	return res
+}
+
+// ExecuteGetOperationStateCommand executes GetOperationsState command
+func ExecuteGetOperationStateCommand(exec CommandExecutorFunc, cmd *RavenCommand) (*GetOperationStateCommandResult, error) {
+	var res GetOperationStateCommandResult
+	err := excuteCmdAndJSONDecode(exec, cmd, &res)
+	if err != nil {
+		return nil, err
+	}
+	return &res, nil
+}
+
+// PutResult describes result of PutDocumentCommand
+// https://sourcegraph.com/github.com/ravendb/ravendb-jvm-client@v4.0/-/blob/src/main/java/net/ravendb/client/documents/commands/batches/PutResult.java#L6
+type PutResult struct {
+	ID           string `json:"Id"`
+	ChangeVector string `json:"ChangeVector"`
+}
+
+func addChangeVectorIfNotEmpty(cmd *RavenCommand, changeVector string) {
+	if changeVector != "" {
+		if cmd.Headers == nil {
+			cmd.Headers = map[string]string{}
+		}
+		cmd.Headers["If-Match"] = fmt.Sprintf(`"%s"`, changeVector)
+	}
+}
+
+// NewPutDocumentJSONCommand creates a command for PutDocument operation
+// TODO: should I validatte js is a valid json?
```

Comment:
Don't have to, the server will do it for you.
Just verify that you have a test for this.

Note that we expect `utf8` encoded json in general